plugin/kubernetes/controller.go

```diff
@@ -144,12 +144,17 @@ func epNameNamespaceIndexFunc(obj interface{}) ([]string, error) {
 	return []string{s.ObjectMeta.Name + "." + s.ObjectMeta.Namespace}, nil
 }
 
-func epIPIndexFunc(obj interface{}) ([]string, error) {
-	ep, ok := obj.(*api.EndpointAddress)
+func epIPIndexFunc(obj interface{}) (idx []string, err error) {
+	ep, ok := obj.(*api.Endpoints)
 	if !ok {
-		return nil, errors.New("obj was not an *api.EndpointAddress")
+		return nil, errors.New("obj was not an *api.Endpoints")
+	}
+	for _, eps := range ep.Subsets {
```

Comment:
Named return params add not much here, it would leave them out. err is not even used.